### ISSUE TITLE
[js] debug classes names

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/haxe#9117]
When compiling to JS using the following options:

```
-D js-classic
-D js-unflatten
```

The classes output is like this:

```js
hide.comp.cdb.Table = $hxClasses["hide.comp.cdb.Table"] = function(editor,sheet,root,mode) { ... }
```

And in the VSCode JS debugger, it shows like this:

![image](https://user-images.githubusercontent.com/1022912/73592095-cec6cc80-44f6-11ea-8034-4f41d1b4a20e.png)

I think that by separating the single assignment into two separates it should display correctly.

**Comment by nadako** [comment]
maybe we should (also?) use named functions for this

**Comment by kLabz** [comment]
This should have been fixed by #10894
